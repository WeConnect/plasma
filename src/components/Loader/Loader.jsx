import React from 'react';
import PropTypes from 'prop-types';
import style from './style.scss';
import {
  getDataAttrs,
  getDataProps,
} from '../../dataUtils';

class Loader extends React.Component {
  render() {
    return (
      <div
        {...getDataAttrs(this.props.data)}
        className={style.container}
        style={this.props.style}
      >
        <div
          className={style.spinner}
        >
          <div
            className={style.bounce1}
            style={this.props.dotStyle}
          />
          <div
            className={style.bounce2}
            style={this.props.dotStyle}
          />
          <div
            className={style.bounce3}
            style={this.props.dotStyle}
          />
        </div>
      </div>
    );
  }
}

Loader.defaultProps = {};

Loader.propTypes = {
  style: PropTypes.object,
  dotStyle: PropTypes.object,
  ...getDataProps(),
};

Loader.displayName = 'Plasma@Loader';

export default Loader;
