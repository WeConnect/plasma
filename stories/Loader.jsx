import React from 'react';
import { storiesOf } from '@storybook/react';
import Loader from '../src/components/Loader/Loader';

import styles from './style.scss';

storiesOf('Loader', module)
  .add('default', () => <Loader />)
  .add('dotColor', () => <Loader dotColor="#ff0000" />)
  .add('dotClassName', () => <Loader dotClassName={styles.greenDotLoader} />);
