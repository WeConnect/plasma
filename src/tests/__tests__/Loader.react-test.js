/* eslint-disable */

import React from 'react';
import Loader from '../../components/Loader/Loader';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

describe('Loader Component', () => {
  test('render', () => {
    const component = renderer.create(
      <Loader />
    );
    expect(component).toMatchSnapshot()
  });
});