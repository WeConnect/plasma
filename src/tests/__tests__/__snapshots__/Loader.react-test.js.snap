// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component render 1`] = `
<div
  className="container"
  style={undefined}
>
  <div
    className="spinner"
  >
    <div
      className="bounce1"
      style={undefined}
    />
    <div
      className="bounce2"
      style={undefined}
    />
    <div
      className="bounce3"
      style={undefined}
    />
  </div>
</div>
`;
